Esri/arcgis-maps-sdk-swift-samples
# [New] Navigate route with rerouting

## Description

This PR implements `Navigate route with rerouting` in `Routing and Logistics` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-NavigateRouteWithReroutingView/Shared/Samples/Navigate%20route%20with%20rerouting)

## Linked Issue(s)

- `swift/issues/4922`

## How To Test

- Press play to start the routing.
- Listen to the voice guidances as the directions update.
- Observe how the route is recalculated when the simulation does not follow the suggested route.
- Tap recenter to reposition the viewpoint. 
- Tap reset to start the simulation from the beginning.

## Screenshots

![navigate-route-with-rerouting](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/6adda6d2-1036-42c9-a382-1835f4e49cfe)
